Use seq to generate the list of times, instead of a bash for-loop

DWesl opened the pull request:
# Description
<!-- This description will become the commit message for the PR-->
<!-- Please use this format for your description:

  Describe your changes. Focus on the *what* and *why*. The *how* will be evident from the changes. In particular, be sure to note any interface changes, such as command line syntax, that will need to be communicated to users.

  At the end of your description, please be sure to add the issue this PR solves using the word "Resolves". If there are any issues that are related but not yet resolved (including in other repos), you may use "Refs".

  Resolves #1234
  Refs #4321
  Refs NOAA-EMC/repo#5678
-->
I'm running a year-long forecast, which means I get a large portion of the log file dedicated to these loops.  

`seq ${START} ${STEP} ${STOP}` will generate a sequence going from START to STOP by STEP, including STOP if relevant.  That seems to be the purpose of these loops.  It will by default separate the list with newlines, but `seq -s ' ' ${START} ${STEP} ${STOP}` will separate them with spaces instead, more closely mimicing the previous behavior.

I would like this to be two lines in the log, rather than a few hundred, and this may also be faster, though probably more for reasons of fewer writes to disk than because bash isn't designed for arithmetic.

# Type of change
<!-- Delete all except one -->
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
  - `$(...)` is a bashism, but the shebang line specifies bash.  Should I switch to backticks instead?
- Does this change require a documentation update? NO

# How has this been tested?
<!-- Please list any test you conducted, including the machine.

Example:
- Clone and build on WCOSS
- Cycled test on Orion
- Forecast-only on Hera
-->
Forecast-only run with these changes on Hera.

# Checklist
- [X] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [X] I have made corresponding changes to the documentation if necessary

DWesl committed:
STY: Use seq to generate forecast hour list instead of for loops

I'm running a year-long forecast, which means I get a large portion of the log file dedicated to these loops.  

`seq ${START} ${STEP} ${STOP}` will generate a sequence going from START to STOP by STEP, including STOP if relevant.  That seems to be the purpose of these loops.

I would like this to be two lines in the log, rather than a few hundred, and this may also be faster, though probably more for reasons of fewer writes to disk than because bash isn't designed for arithmetic.

DWesl committed:
STY:  Have seq separate numbers with spaces, not newlines

This isn't quite as much of the logfile as it was before, but I can make it shorter and more similar to what it was before.

WalterKolczynski-NOAA commented:
@DWesl We converted our shell scripts to strictly bash a while ago, and have forbidden backticks for subshells, so definitely keep using `$(...)`.

WalterKolczynski-NOAA commented on ush/forecast_predet.sh:
```diff
@@ -117,15 +117,10 @@
-    for (( fh = FHMIN; fh < FHMAX_HF; fh = fh + FHOUT_HF )); do
-      FV3_OUTPUT_FH="${FV3_OUTPUT_FH} ${fh}"
-    done
+    FV3_OUTPUT_FH="${FV3_OUTPUT_FH} $(seq -s ' ' ${FHMIN} ${FHOUT_HF} ${FHMAX_HF})"
```
shellcheck is picky
```suggestion
    FV3_OUTPUT_FH="${FV3_OUTPUT_FH} $(seq -s ' ' "${FHMIN}" "${FHOUT_HF}" "${FHMAX_HF}")"
```

WalterKolczynski-NOAA commented on ush/forecast_predet.sh:
```diff
@@ -117,15 +117,10 @@ FV3_predet(){
-    FV3_OUTPUT_FH="${FV3_OUTPUT_FH} ${fh}"
-  done
-
+  FV3_OUTPUT_FH="${FV3_OUTPUT_FH} $(seq -s ' ' ${fhr} ${FHOUT} ${FHMAX})"
```
```suggestion
  FV3_OUTPUT_FH="${FV3_OUTPUT_FH} $(seq -s ' ' "${fhr}" "${FHOUT}" "${FHMAX}")"
```

WalterKolczynski-NOAA reviewed (changes requested):
Always happy to streamline STDOUT.

DWesl committed:
STY: Quote variable dereferences in seq arguments

I would have thought I would need to escape the quotes there.  I don't understand how subshells interact with quoting rules.

Co-authored-by: Walter Kolczynski - NOAA <Walter.Kolczynski@noaa.gov>

WalterKolczynski-NOAA reviewed (approved):
Conditionally approved pending successful completion of CI tests.
